ehrapy/anndata/anndata_ext.py

```diff
@@ -306,6 +306,41 @@ def move_to_x(adata: AnnData, to_x: list[str] | str, copy_x: bool = False) -> An
     return new_adata
 
 
+def get_numerical_column_indices(
+    adata: AnnData, layer: str | None = None, column_indices: Iterable[int] | None = None
+) -> list[int]:
+    adata_layer = adata.X if layer is None else adata[layer]
+    indices = (
+        list(range(adata_layer.shape[1]))
+        if column_indices is None
+        else [i for i in column_indices if i < adata_layer.shape[1] - 1]
+    )
+    non_numerical_indices = []
+    for i in indices:
+        # Todo: There is probably a better way to do this!
+        try:
+            adata_layer[::, i].astype("float64")
+        except ValueError:
+            non_numerical_indices.append(i)
+
+    return [idx for idx in indices if idx not in non_numerical_indices]
+
+
+def get_fully_imputed_column_indices(
+    adata: AnnData, layer: str | None = None, column_indices: Iterable[int] | None = None
+) -> list[int]:
+    adata_layer = adata.X if layer is None else adata.layers[layer]
```

Comment:
```suggestion
    mtx = adata.X if layer is None else adata.layers[layer]
```

adata_layer could also be a layer key. I think we usually call that mtx.